tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/telemetryMetrics/util/InMemoryMetricReader.java

```diff
@@ -0,0 +1,116 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.fault.tolerance.tck.telemetryMetrics.util;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.metrics.InstrumentType;
+import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
+import io.opentelemetry.sdk.metrics.data.LongPointData;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.export.CollectionRegistration;
+import io.opentelemetry.sdk.metrics.export.MetricReader;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.spi.CDI;
+
+@ApplicationScoped
+public class InMemoryMetricReader implements MetricReader {
+
+    private CollectionRegistration collectionRegistration;
+    boolean isShutdown = false;
+
+    public static InMemoryMetricReader current() {
+        return CDI.current().select(InMemoryMetricReader.class).get();
+    }
+
+    @Override
+    public AggregationTemporality getAggregationTemporality(InstrumentType instrumentType) {
+        return AggregationTemporality.CUMULATIVE;
+    }
+
+    @Override
+    public void register(CollectionRegistration registration) {
+        if (isShutdown) {
+            throw new IllegalStateException("InMemoryMetricReader has been shutdown");
+        }
+
+        collectionRegistration = registration;
+    }
+
+    @Override
+    public CompletableResultCode forceFlush() {
+        // TODO Auto-generated method stub
+        throw new UnsupportedOperationException("Unimplemented method 'forceFlush'");
+    }
```

Comment:
Remove TODO

Reply:
It may be that the implementation never calls `forceFlush`, but we should probably fulfil the contract of `MetricReader` properly if we can.

The Javadoc says
> Read and export the metrics.
Our reader doesn't export anything by design, so I think we should succeed immediately here.